packages/pages-shared/src/asset-server/rulesEngine.ts

```diff
@@ -48,9 +48,28 @@ export const generateRulesMatcher = <T>(
 				rule = rule.split(host_match[0]).join(`(?<${host_match[1]}>[^/.]+)`);
 			}
 
-			const path_matches = rule.matchAll(PLACEHOLDER_REGEX);
-			for (const path_match of path_matches) {
-				rule = rule.split(path_match[0]).join(`(?<${path_match[1]}>[^/]+)`);
+			const split = rule.split(/\?(?!<splat>)/g);
```

Comment:
I think this is maybe supposed to be above the `host_matches` piece? Right now, I think you'll incorrectly split at any host placeholder matches e.g. `https://:foo.gregbrimble.com` becomes `https://(?<foo>.*).gregbrimble.com` which, will get split because the negative lookahead only looks for `?<splat>`.

Reply:
It's looking for anything which *isn't* `?<splat>`, which I had to do to get the `/*` headers syntax working. Will look at moving it before the host matching and see if that fixes it.